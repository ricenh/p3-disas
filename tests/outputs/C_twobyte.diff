--- outputs/C_twobyte.txt	2023-11-01 19:33:15.990476456 -0400
+++ expected/C_twobyte.txt	2023-10-10 11:54:35.000000000 -0400
@@ -1,11 +1,8 @@
-Usage: ../y86 <option(s)> mini-elf-file
- Options are:
-  -h      Display usage
-  -H      Show the Mini-ELF header
-  -a      Show all with brief memory
-  -f      Show all with full memory
-  -s      Show the program headers
-  -m      Show the memory contents (brief)
-  -M      Show the memory contents (full)
-  -d      Disassemble code contents
-  -D      Disassemble data contents
+Disassembly of executable contents:
+  0x100:                               | .pos 0x100 code
+  0x100:                               | _start:
+  0x100: 20 01                         |   rrmovq %rax, %rcx
+  0x102: 60 23                         |   addq %rdx, %rbx
+  0x104: a0 6f                         |   pushq %rsi
+  0x106: b0 7f                         |   popq %rdi
+
