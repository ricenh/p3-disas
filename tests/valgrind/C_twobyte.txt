==1601308== Memcheck, a memory error detector
==1601308== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1601308== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1601308== Command: ../y86 -d inputs/twobyte.o
==1601308== 
Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 20 01                         |   rrmovq %rax, %rcx
  0x102: 60 23                         |   addq %rdx, %rbx
  0x104: a0 6f                         |   pushq %rsi
  0x106: b0 7f                         |   popq %rdi

==1601308== 
==1601308== HEAP SUMMARY:
==1601308==     in use at exit: 0 bytes in 0 blocks
==1601308==   total heap usage: 4 allocs, 4 frees, 20,952 bytes allocated
==1601308== 
==1601308== All heap blocks were freed -- no leaks are possible
==1601308== 
==1601308== For lists of detected and suppressed errors, rerun with: -s
==1601308== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
