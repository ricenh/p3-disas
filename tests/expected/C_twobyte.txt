Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 20 01                         |   rrmovq %rax, %rcx
  0x102: 60 23                         |   addq %rdx, %rbx
  0x104: a0 6f                         |   pushq %rsi
  0x106: b0 7f                         |   popq %rdi

